browser/profiles/brave_profile_impl.cc

```diff
@@ -20,6 +21,18 @@ BraveProfileImpl::BraveProfileImpl(
     scoped_refptr<base::SequencedTaskRunner> io_task_runner)
     : ProfileImpl(path, delegate, create_mode, creation_time, io_task_runner),
       weak_ptr_factory_(this) {
+#if !defined(OS_ANDROID)
+  const PrefService::Preference* pref =
+      GetPrefs()->FindPreference(kShieldsAdvancedViewEnabled);
+  if (!pref->HasUserSetting()) {
```

Comment:
I'm thinking which API is more proper (`HasUserSetting()` or `IsDefaultValue()`).
After reading the api's comment, I can't figure out their difference for our purpose. :)
Do you have any specific reason for using `HasUserSetting()`?

Reply:
That's a good question! From the header:
```cpp
// Returns true if the Preference has a user setting, even if that value is
// being overridden by a higher-priority source.
bool HasUserSetting() const;

// Returns true if the Preference is currently using its default value,
// and has not been set by any higher-priority source (even with the same
// value).
bool IsDefaultValue() const;
```

Seems `IsDefaultValue` would work too- since this will return true ONLY if the value is not set 😄 I preferred the `HasUserSetting` because it's the user setting it (or us, on behalf of user)

Reply:
I think both will have same result :)